### PR TITLE
[libra framework] cleanup in RecoveryAddress

### PR DESCRIPTION
**Pull request by sblackshear**
- Fix vector index OOB bug introduced by https://github.com/libra/libra/pull/4595
- Add some new abort codes
- Add functional tests that exercise the abort codes
- Take KeyRotationCapability as input instead of extracting it
- Add comments + document abort behavior in relevant scripts

### CODE REVIEW

File: language/stdlib/modules/RecoveryAddress.move [hunk 1]
```diff
@@ -77,33 +79,33 @@ module RecoveryAddress {
             i = i + 1
         };
         spec {
-            assert i == len + 1;
+            // TODO(shb): Prover thinks this doesn't hold. Is it because of the return in the loop?
+            // How is tis spec block interpreted?
+            //assert i == len;
```
Comment (emmazzz) [inline language/stdlib/modules/RecoveryAddress.move:84]: I think you also need to change line 68 to `assert i <= len;`. Prover relies on the loop invariants to prove line 84.

Comment (sblackshear) [reply language/stdlib/modules/RecoveryAddress.move]: Thanks, that did it!

**Comment by sblackshear** [comment]
Credit to @vgao1996 for noticing the indexing problem!

**Pull request closed without merging**